### ISSUE TITLE
InvalidRequirement when verifying installation

### ISSUE THREAD

**Issue by vs1720** [Lux-AI-Challenge/Lux-Design-S2#221]
Hi! I am trying to set this up locally on my windows machine. I have created a new conda environment based on the environment.yml succesfully however when I run 2 bots against one another as outlined in the readme (the 2 starter ones) I get the following error:

```Traceback (most recent call last):
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\__init__.py", line 3039, in _dep_map
    return self.__dep_map
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\__init__.py", line 2835, in __getattr__
    raise AttributeError(attr)
AttributeError: _DistInfoDistribution__dep_map

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\_vendor\packaging\requirements.py", line 35, in __init__
    parsed = parse_requirement(requirement_string)
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\_vendor\packaging\_parser.py", line 64, in parse_requirement
    return _parse_requirement(Tokenizer(source, rules=DEFAULT_RULES))
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\_vendor\packaging\_parser.py", line 82, in _parse_requirement
    url, specifier, marker = _parse_requirement_details(tokenizer)
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\_vendor\packaging\_parser.py", line 120, in _parse_requirement_details
    specifier = _parse_specifier(tokenizer)
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\_vendor\packaging\_parser.py", line 209, in _parse_specifier
    tokenizer.consume("WS")
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\contextlib.py", line 120, in __exit__
    next(self.gen)
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\_vendor\packaging\_tokenizer.py", line 183, in enclosing_tokens
    self.raise_syntax_error(
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\_vendor\packaging\_tokenizer.py", line 163, in raise_syntax_error
    raise ParserSyntaxError(
pkg_resources.extern.packaging._tokenizer.ParserSyntaxError: Expected closing RIGHT_PARENTHESIS
    opencv-python (>=3.) ; extra == 'all'
                  ~~~~^

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\Scripts\luxai-s2.exe\__main__.py", line 4, in <module>
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\luxai_runner\cli.py", line 8, in <module>
    from luxai_runner.bot import Bot
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\luxai_runner\bot.py", line 10, in <module>
    from luxai_runner.logger import Logger
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\luxai_runner\logger.py", line 1, in <module>
    from luxai_s2.globals import TERM_COLORS
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\luxai_s2\__init__.py", line 2, in <module>
    from .version import __version__
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\luxai_s2\version.py", line 3, in <module>
    __version__ = pkg_resources.require("luxai-s2")[0].version
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\__init__.py", line 956, in require
    needed = self.resolve(parse_requirements(requirements))
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\__init__.py", line 820, in resolve
    new_requirements = dist.requires(req.extras)[::-1]
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\__init__.py", line 2755, in requires
    dm = self._dep_map
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\__init__.py", line 3041, in _dep_map
    self.__dep_map = self._compute_dependencies()
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\__init__.py", line 3051, in _compute_dependencies
    reqs.extend(parse_requirements(req))
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\__init__.py", line 3104, in __init__
    super(Requirement, self).__init__(requirement_string)
  File "C:\Users\Viral\anaconda3\envs\luxai_s2\lib\site-packages\pkg_resources\_vendor\packaging\requirements.py", line 37, in __init__
    raise InvalidRequirement(str(e)) from e
pkg_resources.extern.packaging.requirements.InvalidRequirement: Expected closing RIGHT_PARENTHESIS
    opencv-python (>=3.) ; extra == 'all'
                  ~~~~^```

**Comment by vs1720** [comment]
couldn't find any other issues about this, and I can verify I do have opencv-python installed (which satisfies the version requirement) and I am not sure where this particular string is coming from so I can resolve what is apparently a parenthesis issue?  Would appreciate guidance on this!

**Comment by StoneT2000** [comment]
Could you run conda list

and also share the command you ran?

**Comment by vs1720** [comment]
Command I ran: luxai-s2 kits/python/main.py kits/python/main.py -v 2 -o replay.json

Output of conda list:

```# packages in environment at C:\Users\Viral\anaconda3\envs\luxai_s2:
#
# Name                    Version                   Build  Channel
appdirs                   1.4.4              pyh9f0ad1d_0    conda-forge
brotli                    1.0.9                hcfcfb64_8    conda-forge
brotli-bin                1.0.9                hcfcfb64_8    conda-forge
brotlipy                  0.7.0           py38h91455d4_1005    conda-forge
bzip2                     1.0.8                h8ffe710_4    conda-forge
ca-certificates           2022.12.7            h5b45459_0    conda-forge
certifi                   2022.12.7          pyhd8ed1ab_0    conda-forge
cffi                      1.15.1           py38h57701bc_3    conda-forge
charset-normalizer        2.1.1              pyhd8ed1ab_0    conda-forge
cloudpickle               2.2.1              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
contourpy                 1.0.7            py38hb1fd069_0    conda-forge
cryptography              39.0.1           py38h95f5157_0    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
cython                    0.29.33          py38hd3f51b4_0    conda-forge
fonttools                 4.38.0           py38h91455d4_1    conda-forge
freetype                  2.12.1               h546665d_1    conda-forge
gettext                   0.21.1               h5728263_0    conda-forge
glib                      2.74.1               h12be248_1    conda-forge
glib-tools                2.74.1               h12be248_1    conda-forge
gst-plugins-base          1.22.0               h001b923_0    conda-forge
gstreamer                 1.22.0               h6b5321d_0    conda-forge
gym                       0.21.0           py38h595d716_2    conda-forge
gymnasium                 0.27.1                   pypi_0    pypi
gymnasium-notices         0.0.1                    pypi_0    pypi
icu                       70.1                 h0e60522_0    conda-forge
idna                      3.4                pyhd8ed1ab_0    conda-forge
importlib-metadata        4.13.0                   pypi_0    pypi
importlib-resources       5.12.0             pyhd8ed1ab_0    conda-forge
importlib_resources       5.12.0             pyhd8ed1ab_0    conda-forge
intel-openmp              2023.0.0         h57928b3_25922    conda-forge
jax-jumpy                 0.2.0                    pypi_0    pypi
jpeg                      9e                   hcfcfb64_3    conda-forge
kiwisolver                1.4.4            py38hb1fd069_1    conda-forge
krb5                      1.20.1               heb0366b_0    conda-forge
lcms2                     2.14                 ha5c8aab_1    conda-forge
lerc                      4.0.0                h63175ca_0    conda-forge
libblas                   3.9.0              16_win64_mkl    conda-forge
libbrotlicommon           1.0.9                hcfcfb64_8    conda-forge
libbrotlidec              1.0.9                hcfcfb64_8    conda-forge
libbrotlienc              1.0.9                hcfcfb64_8    conda-forge
libcblas                  3.9.0              16_win64_mkl    conda-forge
libclang                  15.0.7          default_h77d9078_1    conda-forge
libclang13                15.0.7          default_h77d9078_1    conda-forge
libdeflate                1.17                 hcfcfb64_0    conda-forge
libffi                    3.4.2                h8ffe710_5    conda-forge
libglib                   2.74.1               he8f3873_1    conda-forge
libhwloc                  2.9.0                h51c2c0f_0    conda-forge
libiconv                  1.17                 h8ffe710_0    conda-forge
liblapack                 3.9.0              16_win64_mkl    conda-forge
libogg                    1.3.4                h8ffe710_1    conda-forge
libpng                    1.6.39               h19919ed_0    conda-forge
libsqlite                 3.40.0               hcfcfb64_0    conda-forge
libtiff                   4.5.0                hf8721a0_2    conda-forge
libvorbis                 1.3.7                h0e60522_0    conda-forge
libwebp-base              1.2.4                h8ffe710_0    conda-forge
libxcb                    1.13              hcd874cb_1004    conda-forge
libxml2                   2.10.3               hc3477c8_0    conda-forge
libzlib                   1.2.13               hcfcfb64_4    conda-forge
luxai-s2                  2.1.8                    pypi_0    pypi
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
matplotlib                3.7.0            py38haa244fe_0    conda-forge
matplotlib-base           3.7.0            py38h528a6c7_0    conda-forge
mkl                       2022.1.0           h6a75c08_874    conda-forge
msys2-conda-epoch         20160418                      1    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
numpy                     1.24.2           py38h7ec9225_0    conda-forge
opencv-python             4.7.0.68                 pypi_0    pypi
openjpeg                  2.5.0                ha2aaf27_2    conda-forge
openssl                   3.0.8                hcfcfb64_0    conda-forge
packaging                 23.0               pyhd8ed1ab_0    conda-forge
pcre2                     10.40                h17e33f8_0    conda-forge
pettingzoo                1.22.3                   pypi_0    pypi
pillow                    9.4.0            py38h087119c_1    conda-forge
pip                       23.0.1             pyhd8ed1ab_0    conda-forge
ply                       3.11                       py_1    conda-forge
pooch                     1.6.0              pyhd8ed1ab_0    conda-forge
pthread-stubs             0.4               hcd874cb_1001    conda-forge
pthreads-win32            2.9.1                hfa6e2cd_3    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pygame                    2.1.3                    pypi_0    pypi
pyopenssl                 23.0.0             pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.9              pyhd8ed1ab_0    conda-forge
pyqt                      5.15.7           py38hd6c051e_3    conda-forge
pyqt5-sip                 12.11.0          py38hd3f51b4_3    conda-forge
pysocks                   1.7.1              pyh0701188_6    conda-forge
python                    3.8.16          h4de0772_1_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.8                      3_cp38    conda-forge
qt-main                   5.15.8               h720456b_6    conda-forge
requests                  2.28.2             pyhd8ed1ab_0    conda-forge
scipy                     1.10.0           py38h0f6ee2a_2    conda-forge
setuptools                67.3.3             pyhd8ed1ab_0    conda-forge
sip                       6.7.7            py38hd3f51b4_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
tbb                       2021.8.0             h91493d7_0    conda-forge
termcolor                 2.2.0                    pypi_0    pypi
tk                        8.6.12               h8ffe710_0    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tornado                   6.2              py38h91455d4_1    conda-forge
tqdm                      4.64.1             pyhd8ed1ab_0    conda-forge
typing-extensions         4.5.0                    pypi_0    pypi
ucrt                      10.0.22621.0         h57928b3_0    conda-forge
unicodedata2              15.0.0           py38h91455d4_0    conda-forge
urllib3                   1.26.14            pyhd8ed1ab_0    conda-forge
vc                        14.3                hb6edc58_10    conda-forge
vec-noise                 1.1.4                    pypi_0    pypi
vs2015_runtime            14.34.31931         h4c5c07a_10    conda-forge
wheel                     0.38.4             pyhd8ed1ab_0    conda-forge
win_inet_pton             1.1.0              pyhd8ed1ab_6    conda-forge
xorg-libxau               1.0.9                hcd874cb_0    conda-forge
xorg-libxdmcp             1.1.3                hcd874cb_0    conda-forge
xz                        5.2.6                h8d14728_0    conda-forge
yaml                      0.2.5                h8ffe710_2    conda-forge
zipp                      3.14.0             pyhd8ed1ab_0    conda-forge
zstd                      1.5.2                h12be248_6    conda-forge
```

**Comment by vs1720** [comment]
Fixed the issue with a total hack solution. In _parser.py I added: `source = source.replace("opencv-python (>=3.)", "opencv-python (>=3.0)")`

in the parse_requirement function. Leaving this issue open for now in case you have any thoughts as to the root cause of this, but feel free to close it.

**Comment by StoneT2000** [comment]
pip install --upgrade luxai_s2 should fix it, new v2.1.9 version fixes it.
